ompi/mpi/tool/event_get_info.c

```diff
@@ -0,0 +1,137 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2012-2018 Los Alamos National Security, LLC. All rights
+ *                         reserved.
+ * Copyright (c) 2014 Cisco Systems, Inc.  All rights reserved.
+ * Copyright (c) 2017      IBM Corporation. All rights reserved.
+ * Copyright (c) 2018-2019 Triad National Security, LLC. All rights
+ *                         reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "ompi_config.h"
+
+#include "ompi/mpi/tool/mpit-internal.h"
+
+/* needed to convert between opal and ompi datatypes until a function is provided */
+#include "ompi/datatype/ompi_datatype_internal.h"
```

Comment:
Is the bad feeling from including the internal header?  It looks like this can be removed as there is a duplicate hard-coded value of OMPI_DATATYPE_MAX_PREDEFINED in ompi_datatype.h which is previously included through ompi/mpi/tool/mpit-internal.h.

Would deleting these two lines address your concern?

Reply:
Also different defines : OMPI_DATATYPE_MAX_PREDEFINED vs OMPI_DATATYPE_MPI_MAX_PREDEFINED

Reply:
Sorry, needs ompi_datatype_basicDatatypes from ompi_datatypes_internal.h, so you can ignore the above.

Reply:
Alternate proposal as suggested by the comment:

Add a datatype function, such as ompi_datatype_lookup_by_opal_id that provides the ompi_datatype without needed to include ompi_datatype_internal.h.